File: catkin_ws/my_package/scripts/topic_publisher.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int32

def main():
	rospy.init_node('topic_publisher')
	count = 0
	pub = rospy.Publisher('/counter', Int32, queue_size=10)
	rate = rospy.Rate(10) # 10 hz

	while not rospy.is_shutdown():
		pub.publish(count)
		count = count + 1
		rate.sleep()


if __name__ == '__main__':
	main()

File: catkin_ws/src/my_package/src/topic_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"
#include <iostream>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "topic_publisher");
	ros::NodeHandle nh;
	ros::Publisher pub = nh.advertise<std_msgs::Int32>("/counter", 10);
	ros::Rate loop_rate(10);
	int number_count = 0;
	while (ros::ok())
	{
		std_msgs::Int32 msg;
		msg.data = number_count;
		ROS_INFO("%d", msg.data);
		pub.publish(msg);
		ros::spinOnce();
		loop_rate.sleep();
		++number_count;
	}
	return 0;
}